src/realm/object-store/set.cpp

```diff
@@ -284,4 +294,68 @@ std::pair<size_t, bool> Set::insert<Obj>(Obj obj)
     return as<ObjKey>().insert(obj.get_key());
 }
 
+bool Set::is_subset_of(const Set& rhs) const
+{
+    return dispatch([&](auto t) {
+        return this->as<std::decay_t<decltype(*t)>>().is_subset_of(rhs.as<std::decay_t<decltype(*t)>>());
+    });
+}
+
+bool Set::is_superset_of(const Set& rhs) const
+{
+    return dispatch([&](auto t) {
+        return this->as<std::decay_t<decltype(*t)>>().is_superset_of(rhs.as<std::decay_t<decltype(*t)>>());
+    });
+}
+
+bool Set::intersects(const Set& rhs) const
+{
+    return dispatch([&](auto t) {
+        return this->as<std::decay_t<decltype(*t)>>().intersects(rhs.as<std::decay_t<decltype(*t)>>());
+    });
+}
+
+void Set::assign_intersection(const Set& rhs)
+{
+    return dispatch([&](auto t) {
+        return this->as<std::decay_t<decltype(*t)>>().assign_intersection(rhs.as<std::decay_t<decltype(*t)>>());
+    });
+}
+
+void Set::assign_union(const Set& rhs)
+{
+    return dispatch([&](auto t) {
+        return this->as<std::decay_t<decltype(*t)>>().assign_union(rhs.as<std::decay_t<decltype(*t)>>());
+    });
+}
+
+void Set::assign_difference(const Set& rhs)
+{
+    return dispatch([&](auto t) {
+        return this->as<std::decay_t<decltype(*t)>>().assign_difference(rhs.as<std::decay_t<decltype(*t)>>());
+    });
+}
+
 } // namespace realm::object_store
+
+namespace {
+size_t hash_combine()
+{
+    return 0;
+}
+template <typename T, typename... Rest>
+size_t hash_combine(const T& v, Rest... rest)
+{
+    size_t h = hash_combine(rest...);
+    h ^= std::hash<T>()(v) + 0x9e3779b9 + (h << 6) + (h >> 2);
+    return h;
+}
+} // namespace
+
+namespace std {
+size_t hash<realm::object_store::Set>::operator()(realm::object_store::Set const& set) const
```

Comment:
Out of curiosity, what is this used for?

Reply:
The hash function on the Cocoa side